### PR TITLE
Range Enchant and Additive Enchant Fixes

### LINKED ISSUE #67178: FORCEFIELD enchant doesn't seem to work at values below 1
### Describe the bug

While FORCEFIELD, "add": 1 works properly, making the character immune to all damage, lower amounts don't seem to protect to the degree that they should. Even very high values like 0.99 result in a large amount of damage getting through and no attacks blocked.

### Attach save file

[Mindenmines-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/12167413/Mindenmines-trimmed.tar.gz)

### Steps to reproduce

1) Add the following JSON somewhere so it loads properly: 

`  {
    "id": "force_field_generated",
    "type": "effect_type",
    "name": [ "Rippling Air" ],
    "desc": [ "The air around you is wavering strangely." ],
    "apply_message": "The world takes on a slightly bent look, as though you're a straw inserted into a glass of water.",
    "remove_message": "The world snaps back into focus.",
    "rating": "good",
    "max_duration": "1 minutes",
    "enchantments": [ { "condition": "ALWAYS", "values": [ { "value": "FORCEFIELD", "add": 0.999 } ] } ]
  }`

2) Load above game and spawn in with no armor surrounded by hulks
3) Pass one turn while under the effect of Rippling Air, which should make you nearly immune to damage.
4) Explode into chunky salsa

### Expected behavior

Forcefield effectiveness matches the listed numbers

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3208 (22H2)
- Game Version: 88768e1 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    No Rail Stations [No_Rail_Stations],
    Mind Over Matter [mindovermatter]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Bugfixes "non integer additive values for enchants work"

#### Purpose of change
Added a range enchant and resolves #67178

#### Describe the solution
Added enchant
Moved the accessors and data storage for enchants to double

#### Testing
New Perk, tested in game.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
Thanks, it does, I was just going off a mention from discord lol.

File: data/mods/BombasticPerks/perks.json [hunk 1]
```diff
@@ -290,6 +290,15 @@
     "category": [ "perk" ],
     "enchantments": [ { "condition": "ALWAYS", "values": [ { "value": "EVASION", "add": 0.05 } ] } ]
   },
+  {
+    "type": "mutation",
+    "id": "perk_long_shot",
+    "name": { "str": "Long shot" },
```
Comment (TheShadowFerret) [inline data/mods/BombasticPerks/perks.json:296]:
```suggestion
    "name": { "str": "Long Shot" },
```

File: data/mods/BombasticPerks/perks.json [hunk 2]
```diff
@@ -290,6 +290,15 @@
...
+    "points": 0,
+    "description": "You always ate your carrots, that good eyesight is paying off.  You can use guns effectively up to 4 more tiles away.",
```
Comment (TheShadowFerret) [inline data/mods/BombasticPerks/perks.json:298]: This range bonus applies to more than just guns tho 🤔

File: data/mods/BombasticPerks/perks.json (hunk 1 above)
Comment (bombasticSlacks) [reply data/mods/BombasticPerks/perks.json:296]: A bunch aren't fully capitalized, I'll do a standardizing in another PR.

File: data/mods/BombasticPerks/perks.json (hunk 2 above)
Comment (bombasticSlacks) [inline data/mods/BombasticPerks/perks.json:298]:
```suggestion
    "description": "You always ate your carrots, that good eyesight is paying off.  You can use ranged weapons effectively up to 4 more tiles away.",
```

**Pull request merged**